raiden/storage/sqlite.py

```diff
@@ -351,6 +441,89 @@ def _query_events(self, limit: int = None, offset: int = None):
 
         return cursor.fetchall()
 
+    def _get_event_records(
+            self,
+            limit: int = None,
+            offset: int = None,
+            filters: Dict[str, Any] = None,
+    ) -> List[EventRecord]:
+        """ Return a batch of event records
+
+        The batch size can be tweaked with the `limit` and `offset` arguments.
+
+        Additionally the returned events can be optionally filtered with
+        the `filters` parameter to search for specific data in the event data.
+        """
+        limit, offset = _sanitize_limit_and_offset(limit, offset)
+        cursor = self.conn.cursor()
+
+        query = 'SELECT identifier, source_statechange_id, data FROM state_events '
+        where_clauses = []
+        args = []
+        if filters:
```

Comment:
This code can be factored out, since it's used for both state change and filter querying and it does the same thing.

Reply:
Done in https://github.com/raiden-network/raiden/pull/3550/commits/cbbd3c0bcbe804d4f641d88bf8804d140626306f